### PR TITLE
[14.0][IMP] resource_booking: Create resource.is_available() convenience function

### PR DESCRIPTION
**Pull request by carmenbianca**
This work will be a dependency for a module being created by @huguesdk.

Internal task: https://gestion.coopiteasy.be/web#id=9911&action=475&active_id=434&model=project.task&view_type=form&menu_id=536

### CODE REVIEW

**Comment by carmenbianca** [comment]
I'm going to rebase this on top of #77 because I will need this PR to work across days, too.

File: resource_booking/models/resource_booking.py [hunk 1]
```diff
@@ -13,6 +13,38 @@
...
+        end_dt,
+    ):
+        return True
+    # Booking spans more than one calendar day, e.g. from 23:00 to 1:00
+    # the next day.
+    elif (
+        available_intervals
+        and start_date != end_date
+        and (end_date - start_date).days == (len(available_intervals) - 1)
+    ):
```
Comment (carmenbianca) [inline resource_booking/models/resource_booking.py]: We could improve this check by removing it, and iterating over every item to check whether they are on subsequent dates, keeping a rolling tally.

Comment (carmenbianca) [reply resource_booking/models/resource_booking.py]: wrong PR

**Review by huguesdk — APPROVED** [review]
good job reusing `_availability_is_fitting()`!

**Comment by pedrobaeza** [comment]
It seems this one contains more things than the announced in the title.

**Comment by huguesdk** [comment]
it is because it is based on #77. the only actual change is [the last commit](https://github.com/OCA/calendar/pull/80/commits/23caefb0798bca0b4409d4d865b8d6e903ce9ae0).

**Comment by pedrobaeza** [comment]
But for being mergeable, this one should only contain such commit, and I think this one should go before the other.

**Comment by carmenbianca** [comment]
@pedrobaeza I removed the stuff from #77. I'll rebase #77 on top of this.

File: resource_booking/models/resource_booking.py [hunk 2]
```diff
@@ -13,6 +13,17 @@
...
 
+def _availability_is_fitting(available_intervals, start_dt, end_dt):
+    # Booking is uninterrupted on the same calendar day.
+    if (
+        len(available_intervals) == 1
+        and available_intervals._items[0][0] <= start_dt
+        and available_intervals._items[0][1] >= end_dt
+    ):
+        return True
+    return False
```
Comment (pedrobaeza) [inline resource_booking/models/resource_booking.py]:
```suggestion
    return (
        len(available_intervals) == 1
        and available_intervals._items[0][0] <= start_dt
        and available_intervals._items[0][1] >= end_dt
    )
```

File: resource_booking/models/resource_resource.py [hunk 3]
```diff
@@ -14,3 +16,17 @@ def _check_bookings_scheduling(self):
             [("combination_id.resource_ids", "in", self.ids)]
         )
         return bookings._check_scheduling()
+
+    def is_available(self, start_dt, end_dt, domain=None, tz=None):
+        """Convenience method to check whether a resource is available within a
+        time span.
+        """
+        self.ensure_one()
+
```
Comment (pedrobaeza) [inline resource_booking/models/resource_resource.py]: Please avoid empty lines inside a method that makes it less readable. The comment lines already act as separator

File: resource_booking/models/resource_booking.py (hunk 2 above)
Comment (carmenbianca) [reply resource_booking/models/resource_booking.py]: done

File: resource_booking/models/resource_resource.py (hunk 3 above)
Comment (carmenbianca) [reply resource_booking/models/resource_resource.py]: done

**Comment by carmenbianca** [comment]
Thanks for the review @pedrobaeza :man_dancing:

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Pull request merged**